src/catchup/VerifyLedgerChainWork.cpp

```diff
@@ -178,10 +202,15 @@ VerifyLedgerChainWork::verifyHistoryOfSingleCheckpoint()
         if (curr.header.ledgerVersion >
             mApp.getConfig().LEDGER_PROTOCOL_VERSION)
         {
-            return HistoryManager::VERIFY_STATUS_ERR_BAD_LEDGER_VERSION;
+            // Note that local state does not agree with the archives; depending
+            // on the presence of trusted hash
+            mLocalFailure =
+                HistoryManager::VERIFY_STATUS_ERR_BAD_LEDGER_VERSION;
         }
 
         // Verify ledger with local state by comparing to LCL
+        // When checking against LCL, see it the local node is in the bad state,
+        // or if the archive is in a bad state (in which case, retry)
         if (curr.header.ledgerSeq == mLastClosed.first)
```

Comment:
Should be `else if` now that we no longer longer return early above.

Reply:
mm, should it? I guess it depends on what error we want to report first. i was thinking local corrupt state is probably more important to notify the user about, no?

Reply:
yeah I think this is ok -- the second check is possibly-more important to report, and if either happens we're equally sunk.